include/picongpu/fields/currentDeposition/EmZ/DepositCurrent.hpp

```diff
@@ -133,32 +134,35 @@ namespace picongpu
                  * It is done in computeCurrentZ() which has to be explicitly called by a user.
                  * This it different from 3d, where only calling operator() is needed.
                  */
-                template<typename T_Cursor, typename T_Worker>
+                template<typename T_DataBox, typename T_Worker>
                 DINLINE void operator()(
                     T_Worker const& worker,
-                    const T_Cursor& cursorJ,
+                    const T_DataBox& fieldJ,
                     const Line<float2_X>& line,
                     const float_X chargeDensity) const
                 {
-                    using namespace cursor::tools;
-                    cptCurrent1D(worker, cursorJ, line, cellSize.x() * chargeDensity / DELTA_T);
                     cptCurrent1D(
                         worker,
-                        twistVectorFieldAxes<pmacc::math::CT::Int<1, 0>>(cursorJ),
+                        makePermutatedFieldValueAccess<pmacc::math::CT::Int<0, 1>>(fieldJ),
+                        line,
+                        cellSize.x() * chargeDensity / DELTA_T);
+                    cptCurrent1D(
+                        worker,
+                        makePermutatedFieldValueAccess<pmacc::math::CT::Int<1, 0>>(fieldJ),
```

Comment:
What is this doing?

Reply:
I added docu to `makePermutatedFieldValueAccess`